rfcs/chromium-connector.md

```diff
@@ -0,0 +1,213 @@
+| JS evaluation         | [`page.evaluate(pageFunction[, ...args])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pageevaluatepagefunction-args) |
+| Request notification  | [`page.on('request')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-request) <br> [`page.on('requestfailed')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-requestfailed) <br> [`page.on('requestfinished')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-requestfinished) |
+| Page errors | [`page.on('error)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-error) <br> [`page.on('pageerror)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-pageerror) (for uncaught exceptions) |
+| Response notification | [`page.on('response')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-response) |
+| Close browser         | [`browser.close()`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-browserclose) |
+| Navigate              | [`browser.newPage()`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-browsernewpage) <br> [`page.goto(url[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagegotourl-options) <br> [`page.waitForNavigation([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagewaitfornavigationoptions) |
+| Set page content | [`page.setContent(html[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagesetcontenthtml-options) |
+| Input | [`page.type(selector, text[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagetypeselector-text-options) <br> [`page.click(selector[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pageclickselector-options) |
+
+
+## Other comments/questions
+
+**How to run multiple pages in the same browser from different Puppeteer instances?**
+
+The idea is to use a similar approach where the `PID` is stored in a file and alternate between [`puppeteer.launch([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerlaunchoptions) and [`puppeteer.connect(options)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerconnectoptions) when the browser is already running (using `browserWSEndpoint`).
+
+**How to run a different Chromium based browser?**
+
+[`puppeteer.launch([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerlaunchoptions) has an option to pass `executablePath`. An option should be exposed in `connector-chromium` configuration to specify it. This value could be the `path` or maybe a well-known `string` value (`chromium-canary`, `chrome`, etc.) and someone (CLI, `connetor`?) could resolve the path similar to what `chrome-launcher` does today to decide what browser to run.
```

Comment:
Assuming we allow for per-connector configuration types, should `connector-chromium` allow passing forward any options supported by `puppeteer`?

Reply:
Yes, we already do that with the `chrome-connector` and `chrome-launcher`. It's just that finding the path to the executable might be problematic and we can probably improve the ux and have some well-known folders to look into so they can just decide.